transbank/onepay/transaction.py

```diff
@@ -0,0 +1,122 @@
+# encoding: utf-8
+import calendar
+import http.client
+import json
+
+from datetime import datetime
+from enum import Enum
+from urllib.parse import urlparse
+
+from transbank.onepay.schema import ItemSchema, TransactionCreateRequestSchema, TransactionCreateResponseSchema, SendTransactionResponseSchema
+
+from transbank.onepay.cart import ShoppingCart
+from transbank.onepay.error import TransactionCreateError, SignError
+from transbank.onepay import sign
+
+from transbank import onepay
+
+class Channel(Enum):
+    WEB = "WEB"
+    MOBILE = "MOBILE"
+    APP = "APP"
+
+class Options(object):
+    def __init__(self, api_key: str, shared_secret: str):
+        self.api_key = api_key
+        self.shared_secret = shared_secret
+
+class TransbankRequestResponse(object):
+    signable_attributes = []
+
+    def get_signable_data(self, append_data = []):
+        signable_data = [getattr(self, item) for item in self.signable_attributes]
+        return signable_data + append_data
+
+class TransactionCreateRequest(TransbankRequestResponse):
+
+    signable_attributes = ['external_unique_number', 'total', 'items_quantity', 'issued_at']
+
+    def __init__(self, external_unique_number, total, items_quantity, issued_at, items,
+                 callback_url = None, channel = "WEB", app_scheme = None, options = None):
+
+        self.external_unique_number = external_unique_number
+        self.total = total
+        self.items_quantity = items_quantity
+        self.issued_at = issued_at
+        self.items = items
+        self.callback_url = callback_url
+        self.channel = channel
+        self.app_scheme = app_scheme
+        self.app_key = onepay.integration_type.value.app_key
+        self.api_key = (options or onepay).api_key
+        self.generate_ott_qr_code = True
+        self.options = options or onepay
+
+    @property
+    def signature(self):
+        return sign.build_signature_for_transaction_create_request(self, self.options.shared_secret)
+
+class TransactionCreateResponse(TransbankRequestResponse):
```

Comment:
And here we wouldn't need anything extra. The `is_valid_signature()` method from `Signable` does the work perfectly for us here.